sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -486,6 +486,12 @@ object PartitioningUtils extends SQLConfHelper{
       DateType
     }
 
+    val binaryTry = Try {
+      val binaryValue = Hex.unhex(raw.getBytes())
```

Comment:
Wouldn't this infer binary type virally from all the string types?

Reply:
I will revert this change. Sorry I just realized this PR is not complete yet. I have converted this to a draft RP.